src/iuse.cpp

```diff
@@ -2887,17 +2967,47 @@ int iuse::siphon( player *p, item *it, bool, const tripoint & )
         p->add_msg_if_player( m_info, _( "You cannot do that while mounted." ) );
         return 0;
     }
-    const cata::optional<tripoint> pnt_ = choose_adjacent( _( "Siphon from where?" ) );
-    if( !pnt_ ) {
-        return 0;
+    const std::function<bool( const tripoint & )> f = []( const tripoint & pnt ) {
+        const optional_vpart_position vp = g->m.veh_at( pnt );
+        return !!vp;
+    };
+
+    vehicle *v = nullptr;
+    bool found_more_than_one = false;
+    for( const tripoint &pos : g->m.points_in_radius( g->u.pos(), 1 ) ) {
+        const optional_vpart_position vp = g->m.veh_at( pos );
+        if( !vp ) {
+            continue;
+        }
+        vehicle *vfound = &vp->vehicle();
+        if( v == nullptr ) {
+            v = vfound;
+        } else {
+            //found more than one vehicle?
+            if( *v != *vfound ) {
```

Comment:
```suggestion
            if( v != vfound ) {
```
And get rid of the `operator==`  for vehicles. That operator does not behave intuitive and in fact is wrong: `vehicle::om_id` is *only* set when the vehicle is marked to be remember by a player character, otherwise it's uninitialized. And vehicles on different overmaps may have the very same `om_id` as well.

Comparing the pointer is the way as vehicle pointers are unique (and it's done that way in other parts of the code).